apps/prairielearn/src/ee/lib/validateHTML.ts

```diff
@@ -0,0 +1,616 @@
+import * as parse5 from 'parse5';
+
+const HTML_GLOBAL_TAGS = [
+  'accesskey',
+  'class',
+  'contenteditable',
+  'data-*',
+  'dir',
+  'draggable',
+  'enterkeyhint',
+  'hidden',
+  'id',
+  'inert',
+  'inputmode',
+  'lang',
+  'popover',
+  'spellcheck',
+  'style',
+  'tabindex',
+  'title',
+  'translate',
+];
+
+/**
+ * Checks that the required property is an int (or property to be inserted) or adds an error to the provided list.
+ * @param tag The name of the tag being checked.
+ * @param key The property name.
+ * @param val The property value.
+ * @param errors The list of errors to add to.
+ */
+function assertInt(tag: string, key: string, val: string, errors: string[]) {
+  if (!(/^\d+$/.test(val) || /^\{\{.*\}\}$/.test(val))) {
```

Comment:
Nit: consider lifting the Mustache regexp up into a constant so that we can document what it is (it took me a moment to realize why this was here).

Reply:
Changed in 0e87e7e4333c2d4d8dfbc2116aad0c0acd81707b.